core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AdaptivePlanOptimizer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.sql.planner.optimizations;
+
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public interface AdaptivePlanOptimizer
+        extends PlanOptimizer
+{
+    @Override
+    default PlanNode optimize(PlanNode plan, Context context)
+    {
+        return optimizeAndMarkPlanChanges(plan, context).plan();
+    }
+
+    /**
+     * Optimize the plan and return the changes made to the plan. This method is needed for adaptive
```

Comment:
Maybe drop  from here`This method is needed for adaptive optimization in FTE.`  but add general this as general information for `AdaptivePlanOptimizer` class.